extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -661,6 +664,59 @@ class LifecycleTask extends BackbeatTask {
         return false;
     }
 
+    /**
+     * Helper method for Expiration.ExpiredObjectDeleteMarker rule
+     * Check if ExpiredObjectDeleteMarker rule applies to the `IsLatest` delete
+     * marker
+     * @param {object} bucketData - bucket data
+     * @param {object} deleteMarker - single non-current version
+     * @param {string} deleteMarker.LastModified - last modified date
+     * @param {object} rules - most applicable rules from `_getApplicableRules`
+     * @param {Logger.newRequestLogger} log - logger object
+     * @param {function} done - callback(error)
+     * @return {undefined}
+     */
+    _checkAndApplyEODMRule(bucketData, deleteMarker, rules, log, done) {
+        return this._listVersions(bucketData, { Prefix: deleteMarker.Key }, log,
```

Comment:
The only edge case I can think of is that the listing was just truncated and this key is the last version/delete marker in the listing. 

I was thinking I could pass a flag or some indicator to tell if this key is alone. One thing I can do is if this is the last version/delete marker, it will match with the `NextVersionIdMarker` and `NextKeyMarker`, which can serve as an indicator to make this prefix listing

Reply:
sounds good to me